force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+        Test.stopTest();
+
+        System.assertEquals(false, isAddressEmpty);
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isContactAccAddrInfoSame ******************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void isContactAccAddrInfoSameTrue() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameFalseDifferentAddressField() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+        contact.MailingCity = 'Something else';
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameFalseDifferentStateCode() {
+        /*Almost no way we can test state code/country code here
+         This need to be refactored so that we can override the returned result*/
+        //TODO(Refactor): Make isStateCountryPicklistsEnabled overridable
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameNotNullContactNullAccount() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, null);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameNullContactNotNullAccount() {
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(null, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameNullContactNullAccount() {
+        //We don't have guard clause for the scenario when both contact and account are null
```

Comment:
If this is the case, make a test for this that covers the scenario and asserts for the current behavior. Then make a comment in the ADDR_Addresses_UTIL to highlight the problem scenario.

Reply:
You mean catch the null pointer and assert the exception type?

Reply:
Yes